src/Storages/WindowView/StorageWindowView.cpp

```diff
@@ -1415,6 +1414,8 @@ void StorageWindowView::writeIntoWindowView(
 
 void StorageWindowView::startup()
 {
+    DatabaseCatalog::instance().addDependency(select_table_id, getStorageID());
```

Comment:
I am not sure, need to check that `shutdown` is called only if `startup` was called as we might not call `startup` because of some exception and then `removeDependency` in `shutdown` will fail.

Reply:
Here is a document but only explain the execution order of `startup` and `shutdown`, and it is not stated that `shutdown` will called only if `startup` was called.
https://github.com/ClickHouse/ClickHouse/blob/82c30442bdafbf9464551dae3812fb9486153455/src/Interpreters/InterpreterCreateQuery.cpp#L1300-L1311

So I have dig into `StorageFactory::instance().get`, except system|information tables, it only called by `createTableFromAST` and `InterpreterSelectQuery`, the `createTableFromAST` will called by

1. `DatabaseLazy`, follows `startup()`
2. `DatabaseOrdinary::tryAttachTable`->`loadTableFromMetadata()`, follows `startup`
3. `DatabaseCatalog::enqueueDroppedTableCleanup`, do not startup and shutdown, only mark `is_dropped` variable